sklearn/compose/_column_transformer.py

```diff
@@ -394,6 +394,34 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name))
 
+    def _validate_features(self, n_features, feature_names):
+        """Ensures feature counts and names are the same during fit and
+        transform.
+
+        TODO: It should raise an error from v0.24
+        """
+
+        if ((self._feature_names_in is None or feature_names is None)
+                and self._n_features == n_features):
+            return
+
+        neg_col_present = np.any([_is_negative_indexing(col)
+                                  for col in self._columns])
+        if neg_col_present and self._n_features != n_features:
+            raise RuntimeError("At least one negative column was used to "
```

Comment:
Just curious about RuntimeError vs ValueError?

Reply:
When I was typing, I first had a `ValueError`, but it kinda seems more like a `RuntimeError` to me. I don't have a strong feeling on the type of the error, it was a gut feeling. Happy to change if you think the other one is more apt.